java/vector/src/test/java/org/apache/arrow/vector/TestBufferOwnershipTransfer.java

```diff
@@ -60,8 +60,8 @@ public void testTransferVariableidth() {
 
     NullableVarCharVector v1 = new NullableVarCharVector("v1", childAllocator1);
     v1.allocateNew();
-    v1.getMutator().setSafe(4094, "hello world".getBytes(), 0, 11);
```

Comment:
Is `getMutator` removed from the new API?

Reply:
Yes, one of the goals is to simplify the access and reduce the number of objects/classes. So all the getters and setters are directly accessible.

Reply:
Ok, the existing Spark code won' compile with this then. @BryanCutler or I can make the change in Spark, but in general I wonder if this is too disruptive.

Reply:
Doubling the number of getter/setter API is also not great, maybe we can add deprecated method and remove them in the next two releases or after 0.8 or sth. This is more a question for @wesm and @jacques-n to if we are willing to break user code in 0.8.

Reply:
I think the potential plan is to have the following:
Legacy Nullable <Type> Vector
Nullable <type> Vector -- the ones being implemented as part of refactoring

The latter one is not mutator/accessor based.

The existing consumers of these Nullable vectors should be able to continue using them through the LegacyNullable vector which is merely a delegation to new code for each API.

For example, if in Dremio code, we replace each reference to NullableIntVector with LegacyNullableIntVector, the code should still work without any surprises.

We can keep the legacy code around for now and then remove them all together (along with NullableValueVectors template) in the next release or so.

Reply:
After thinking about this a bit more, I feel maintaining backward compatibility of the new vectors might not to possible. The reasons are because I think we want to rename some vectors properly:

* DateDayVector, DateMillisVector ->DateVector
* IntervalDayVector -> IntervalVector
* Timestamp*Vector -> TimestampVector
* MapVector -> StructVector

Given this, we might as well just introduce new classes of vectors that is are slightly different (for instance, no accessor/mutator)

Also, I assume we would want ArrowReaders to return VectorSchemaRoot with new classes of vector in 0.8, is that correct?

Also,  ArrowWriters should work with both legency and new vectors?

Reply:
Btw, I updated the requirement doc and added the renaming requirement.

Reply:
+1 on renaming. Are we also planning on removing the "Nullable" prefix to vectors?  like `NullableIntVector` -> `IntVector`.  I know that's currently what the inner vectors are called but after the refactor it won't really make sense to call it `NullableIntVector` because the inner vectors will be gone and all vectors are nullable right?

Reply:
Yes, that's probably right. @jacques-n , what do you think?